src/main/java/com/exasol/adapter/document/mapping/PropertyToVarcharColumnValueExtractor.java

```diff
@@ -84,18 +91,27 @@ private String handleOverflow(final String tooLongSourceString) {
             return tooLongSourceString.substring(0, this.column.getVarcharColumnSize());
         } else {
             throw new OverflowException(
-                    "String overflow. You can either increase the string size if this column or set the overflow behaviour to truncate.",
+                    ExaError.messageBuilder("E-VSD-38").message(
+                            "A value for column {{COLUMN}} exceeded the configured varcharColumnSize of {{SIZE}}.")
+                            .parameter("COLUMN", this.column.getExasolColumnName())
+                            .parameter("SIZE", this.column.getVarcharColumnSize())
+                            .mitigation("Increase 'varcharColumnSize' for this column.")
+                            .mitigation("Set 'overflowBehaviour' for this column to 'ABORT' or 'NULL'.").toString(),
                     this.column);
         }
     }
 
     /**
-     * This class is used to pass the result of {@link #mapStringValue(DocumentNode)} from the concrete implementation
-     * to this abstract class.
-     * 
+     * This interface is used to pass the result of {@link #mapStringValue(DocumentNode)} from the concrete
+     * implementation to this abstract class.
+     *
      * @implNote public so that it is accessible from test code
      */
-    public static class MappedStringResult {
+    public interface ConversionResult {
```

Comment:
Because they are different. The one is for the ToDecimal conversion result, the other for the to string conversion result. 
Since this are different operations they have different return types and different errors.
The fully qualified name is by the way different since these are nested classes.